java/src/com/ibm/streamsx/topology/internal/context/streamsrest/EdgeImageContext.java

```diff
@@ -132,10 +135,12 @@ protected void postBuildAction(JsonObject deploy, JsonObject jco, JsonObject res
             
             if (! "built".equals(buildStatus)) {
                 TRACE.severe("The image failed to build with status " + buildStatus + ".");
-                // TODO: Make error handling regarding the result when postBuildAction(...) in DistributedStreamsRestContext fails?
+                throw new IllegalStateException("Error submitting bundle for build edge image: " + buildName);
```

Comment:
Here, the log messages should be retrieved from build service and traced. Similar to Build.java lines 175...182:
```
StreamsRestUtils.TRACE.severe("The submitted archive " + archive.getName() + " failed to build with status " + getStatus() + ".");
		
Request gr = Request.Get(this.logMessages).addHeader("Authorization", connection().getAuthorization());
				
String output = StreamsRestUtils.requestTextResponse(connection().executor, gr);
String[] lines = output.split("\\R");
for (String line : lines)
    StreamsRestUtils.TRACE.severe(line);
```
The code cannot be simply copied as it references the package private class `StreamsRestUtils`. Instead, the code portion in Build.java should be extracted into a separate method `public List<String> getLogMessages()` and be used in Build.java and here.
This can be done in a different PR, however.

Reply:
yes, will be done in upcoming PR